autoray/lazy/core.py

```diff
@@ -1171,6 +1186,16 @@ def split(ary, indices_or_sections, axis=0):
     return tuple(sub_arys)
 
 
+def where(condition, x, y):
+    x = ensure_lazy(x)
+    condition = ensure_lazy(condition)
+    return x.to(
+        fn=get_lib_fn(x.backend, "where"),
+        args=(condition, x, y),
+        deps=tuple(a for a in (condition, x, y) if isinstance(a, LazyArray))
+    )
```

Comment:
This looks good for now. Eventually it might be nice to support the full combination of args, e.g. broadcasting `x`. One might then have to use `find_common_backend` and `find_broadcast_shape` and not use `x.to`

Reply:
Done, used the find_common_backend and find_broadcast_shape for the broadcasting case.